CompilerForCAP/tst/CapJitResolvedOperations.tst

```diff
@@ -19,16 +19,19 @@ gap> cat := DummyCategory( rec( list_of_operations_to_install := [ "IsCongruentF
 gap> StopCompilationAtPrimitivelyInstalledOperationsOfCategory( cat );
 gap> Display( CapJitCompiledFunction( func, cat, [ "category", "morphism", "morphism" ], "morphism" ) );
 function ( cat_1, mor1_1, mor2_1 )
-    return PreComposeList( cat_1, [ mor1_1, mor2_1 ] );
+    return PreComposeList( cat_1, Source( mor1_1 ), [ mor1_1, mor2_1 ], Range( mor2_1 ) );
 end
 
 # resolve CAP operations without category as first argument (list input)
-gap> func := { cat, mor1, mor2 } -> PreComposeList( [ mor1, mor2 ] );;
-gap> cat := DummyCategory( rec( list_of_operations_to_install := [ "IsCongruentForMorphisms", "PreCompose" ] ) );;
+gap> func := { cat, mor1, mor2 } -> PreComposeList( Source( mor1 ), [ mor1, mor2 ], Range( mor2 ) );;
```

Comment:
This example should test if CompilerForCAP can derive the category from a list. This is not the case anymore because the category can now be derived from the new objects, which is why the coverage in `CompilerForCAP/gap/ResolveOperations.gi` drops. Please refactor this example (using `DirectProduct` derived from `DirectSum` could work).

Reply:
I hope I understood you correctly. I had to use `IsomorphismFromDirectProductToDirectSum` instead of `DirectSum`.